spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc

```diff
@@ -164,6 +188,86 @@ Make sure to add 'spring-boot-configuration-processor' as an optional dependency
 ----
 ====
 
+
+[[spring-cloud-dataflow-stream-app-metadata-artifact]]
+=== Creating and using a dedicated metadata artifact
+You can go a step further in the process of describing the main properties that your stream or task app supports by
+creating a so-called metadata companion artifact. This simple jar file contains only the Spring boot JSON file about
+configuration properties metadata, as well as the whitelisting file described in the previous section.
+
+Here is the contents of such an artifact, for the canonical `log` sink:
+
+[source, bash]
+----
+$ jar tvf log-sink-rabbit-1.2.0.BUILD-SNAPSHOT-metadata.jar
+373848 META-INF/spring-configuration-metadata.json
+   174 META-INF/spring-configuration-metadata-whitelist.properties
+----
+
+Note that the `spring-configuration-metadata.json` file is quite large. This is because it contains the concatenation of _all_ the properties that
+are available at runtime to the `log` sink (some of them come from `spring-boot-actuator.jar`, some of them come from
+`spring-boot-autoconfigure.jar`, even some more from `spring-cloud-starter-stream-sink-log.jar`, _etc._) Data Flow
+always relies on all those properties, even when a companion artifact is not available, but here all have been merged
+into a single file.
+
+To help with that (as a matter of fact, you don't want to try to craft this giant JSON file by hand), you can use the
+following plugin in your build:
+
+[source, xml]
+----
+<plugin>
+ 	<groupId>org.springframework.cloud</groupId>
+ 	<artifactId>spring-cloud-app-starter-metadata-maven-plugin</artifactId>
+ 	<executions>
+ 		<execution>
+ 			<id>aggregate-metadata</id>
+ 			<phase>compile</phase>
+ 			<goals>
+ 				<goal>aggregate-metadata</goal>
+ 			</goals>
+ 		</execution>
+ 	</executions>
+ </plugin>
+----
+
+NOTE: This plugin comes in _addition_ to the `spring-boot-configuration-processor` that creates the individual JSON files.
+Be sure to configure the two!
+
+The benefits of a companion artifact are manifold:
+
+1. being way lighter (usually a few kilobytes, as opposed to megabytes for the actual app), they are quicker to download,
```

Comment:
Perhaps it might be beneficial to also list the GUI experience either in the same or separate bullet. That's where we have had folks reporting UX issues.